source/common/http/http1/codec_impl.cc

```diff
@@ -351,18 +358,31 @@ static const char REQUEST_POSTFIX[] = " HTTP/1.1\r\n";
 void RequestEncoderImpl::encodeHeaders(const RequestHeaderMap& headers, bool end_stream) {
   const HeaderEntry* method = headers.Method();
   const HeaderEntry* path = headers.Path();
-  if (!method || !path) {
+  const HeaderEntry* host = headers.Host();
+  bool is_connect = HeaderUtility::isConnect(headers);
+
+  if (is_connect && !host) {
+    throw CodecClientException("Host must be specified for CONNECT requests");
+  } else if (!method || (!path && !is_connect)) {
```

Comment:
This came up in @asraa's exception removal PR, but these exceptions have no purpose. I don't recall the history of why they are here but they are programming errors and will crash the server anyway, so perhaps just switch them to RELEASE_ASSERT or ASSERT now to avoid @asraa more hassle later? Up to you.

Reply:
What did we decide here? Do you want to just ASSERT or RELEASE_ASSERT the new case vs. throw an exception?